src/components/Chart/Table.js

```diff
@@ -0,0 +1,255 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+
+import { css } from 'glamor'
+import { scaleThreshold, scaleQuantize, scaleOrdinal } from 'd3-scale'
+import { extent } from 'd3-array'
+
+import { useColorContext } from '../Colors/ColorContext'
+import { getFormat, getTextColor } from './utils'
+import { ExpandMoreIcon, ExpandLessIcon } from '../Icons'
+import { defaultProps } from './ChartContext'
+import { sansSerifRegular18 } from '../Typography/styles'
+
+const styles = {
+  container: css({
+    overflowX: 'auto',
+    overflowY: 'hidden',
+    marginLeft: '-20px',
+    marginRight: '-20px'
+  }),
+  table: css({
+    ...sansSerifRegular18,
+    lineHeight: '1.2',
+    minWidth: '100%',
+    borderSpacing: '0 2px',
+    borderCollapse: 'separate',
+    padding: '0 10px'
+  }),
+  header: css({
+    borderBottomWidth: '1px',
+    borderBottomStyle: 'solid',
+    padding: '8px 0',
+    userSelect: 'none',
+    '&:first-of-type': {
+      padding: '8px 0 8px 10px'
+    },
+    '&:last-of-type': {
+      padding: '8px 10px 8px 0'
+    }
+  }),
+  cell: css({
+    padding: '8px 0',
+    '&:first-of-type': {
+      padding: '8px 0 8px 10px'
+    },
+    '&:last-of-type': {
+      padding: '8px 10px 8px 0'
+    }
+  })
+}
+
+const Table = props => {
+  const [colorScheme] = useColorContext()
+  const {
+    values,
+    numberFormat,
+    colorRanges,
+    colorRange,
+    defaultSortColumn,
+    thresholds,
+    tableColumns
+  } = props
+  const columns = values.columns
+
+  const [sortBy, setSortBy] = useState({
+    key: defaultSortColumn,
+    order: 'desc',
+    enableSorting: !!defaultSortColumn || false
+  })
+
+  let parsedData = []
+
+  values.forEach(row => {
+    let parsedItem = {}
+    Object.keys(row).forEach(
+      item =>
+        (parsedItem[item] =
+          tableColumns.find(d => d.column === item)?.type === 'number'
+            ? +row[item]
+            : row[item])
+    )
+    parsedData.push({ ...parsedItem })
+  })
+
+  const sortedData = parsedData.sort((a, b) => {
+    if (!sortBy.enableSorting) {
+      return parsedData
+    }
+    if (typeof a[sortBy.key] === 'string') {
+      return sortBy.order === 'desc'
+        ? a[sortBy.key].localeCompare(b[sortBy.key])
+        : b[sortBy.key].localeCompare(a[sortBy.key])
+    }
+    return sortBy.order === 'desc'
+      ? b[sortBy.key] - a[sortBy.key]
+      : a[sortBy.key] - b[sortBy.key]
+  })
```

Comment:
`sort` mutates the array – you shouldn't create a new variable because the old one is also changes anyhow.

Also always use the d3-array functions:
```suggestion
  if (sortBy.key) {
    parsedData.sort((a, b) => {
      return sortBy.order === 'desc'
        ? descending(a[sortBy.key], b[sortBy.key]),
        : ascending(a[sortBy.key], b[sortBy.key])
    })
  }
```
or is there something wrong with them?